bcgov/HOUS-permit-portal
# BPHH-1388 Add more UI details for submitter diff

## Description

Move compare logic to versioning service
Add requirement highlighting for specific changes in form for submitter
Move diff variables into permit application mst model
start work on elective notifications

## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature

## Related Tickets & Documents

https://hous-bssb.atlassian.net/browse/BPHH-1388

## Steps to QA

log in as submitter in regular window
log in as super admin in incognito window
create permit application as submitter

as super admin, edit the corresponding requirement template, adding, removing, and changing requirements in different sections

notification should generate for submitter

refresh permit application form page in submitter window, yellow diff box should display
see that the add, removed, and changes are in their own sections
these sections are also additionally broken down by section label
Clicking links should scroll to appropriate location in form and bold the text

log in as review manager, see the latest notification and click go
verify that the compare box is still present and correct, and that the scroll links still work
